zeus/device/gpu.py

```diff
@@ -271,31 +273,6 @@ def getTotalEnergyConsumption(self) -> int:
 
 
 def _handle_nvml_errors(func):
-    @functools.wraps(func)
-    def wrapper(*args, **kwargs):
-        try:
-            return func(*args, **kwargs)
-        except pynvml.NVMLError as e:
-            exception_class = NVIDIAGPU._exception_map.get(e.value, ZeusGPUUnknownError)
-            raise exception_class(e.msg) from e
-
-    return wrapper
-
-
-class NVIDIAGPU(GPU):
-    """Control a Single NVIDIA GPU.
-
-    Uses NVML Library to control and query GPU. There is a 1:1 mapping between the methods in this class and the NVML library functions.
-    Zeus GPU Exceptions are raised when NVML errors occur.
-    To ensure computational efficiency, this class utilizes caching (ex. saves the handle) to avoid repeated calls to NVML.
-    """
-
-    def __init__(self, gpu_index: int) -> None:
-        """Initializes the NVIDIAGPU object with a specified GPU index. Acquires a handle to the GPU using `pynvml.nvmlDeviceGetHandleByIndex`."""
-        super().__init__(gpu_index)
-        self._get_handle()
-        self._supportsGetTotalEnergyConsumption = None
-
     _exception_map = {
```

Comment:
This should be a global variable, otherwise it'll be constructed from scratch on every call to `_handle_nvml_errors`.